compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/GPUTileSwizzleUtils.cpp

```diff
@@ -398,6 +400,86 @@ TileSwizzle getSwizzle(IREE::GPU::DataTiledMMAAttr mma,
   return crossThreadOuterSwizzle;
 }
 
+/// Remove the expanded dimensions for this index and update the permutation by
+/// erasing the removed dimensions' indices and adjusting existing larger
+/// indices accordingly.
+static void remove(TileSwizzle &swizzle, size_t idx) {
+  assert(idx < swizzle.expandShape.size() && "idx out of bounds");
+  size_t startIdx = std::accumulate(
+      std::begin(swizzle.expandShape), std::begin(swizzle.expandShape) + idx, 0,
+      [](size_t idx, const TileSwizzle::ExpandShapeDimVectorType &dims)
+          -> size_t { return idx + dims.size(); });
```

Comment:
I think this can be simplified to:
```suggestion
  size_t startIdx = swizzle.expandShape[idx].front();
```
The dims in the swizzle should always be in order because they represent an expand_shape reassociation.

Reply:
`expandShape` is something like `[[4, 8, 4], [4, 4]]`. We don't need the size at the front of dimension `idx` (`4` in this case), but the index in case this `expandShape` would be flattened. For example:
```
idx == 0 -> startIdx == 0
idx == 1 -> startIdx == 3
```

Reply:
Ah, right. I forgot we use the expandShape to hold the sizes. Ignore my comment then.